src/validation/validateProperties.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT
+ */
+
+import {
+    Diagnostic,
+    ObjectASTNode,
+    TextDocument,
+} from 'vscode-json-languageservice';
+
+import {
+    findPropChildByName,
+    isArrayNode,
+    isObjectNode,
+} from '../utils/astUtilityFunctions'
+
+import {
+    DiagnosticsForNodeFunc,
+    getDiagnosticsForArrayOfSchema,
+    getDiagnosticsForOneOfSchema,
+    getDiagnosticsForRegularProperties,
+    SchemaObject,
+} from './utils/validatePropertiesUtils'
+
+import schema from './validationSchema'
+
+const referenceTypes = schema.ReferenceTypes
+
+const getDiagnosticsForNode: DiagnosticsForNodeFunc = function(rootNode, document, schemaPart) {
```

Comment:
I suggest moving `getDiagnosticsForNode` this into `validatePropertiesUtils.ts` and making it the only export. The helpers for the different sub-schema types don't need to be public.

This will get rid of the `DiagnosticsForNodeFunc` interface and having to pass `getDiagnosticsForNode` as a parameter.

Reply:
done